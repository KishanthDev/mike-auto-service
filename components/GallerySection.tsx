"use client";

import React from "react";
import Image from "next/image";
import images from "../data/gallery.json";

export default function GallerySection() {
  return (
    <section>
      <h2 className="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#1a3c6e] dark:text-white">
        Our Shop Gallery
      </h2>
      <div className="grid grid-cols-2 sm:grid-cols-4 gap-3 md:gap-4 mt-4 md:mt-6">
        {images.map((src, idx) => (
          <div
            key={idx}
            className="aspect-square overflow-hidden rounded-lg bg-gray-100 dark:bg-gray-700 shadow-md"
          >
            <Image
              src={src}
              alt={`Gallery ${idx + 1}`}
              width={200}
              height={200}
              className="w-full h-full object-cover transition-transform hover:scale-105"
            />
          </div>
        ))}
      </div>
    </section>
  );
}
