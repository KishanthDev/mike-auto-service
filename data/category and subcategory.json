[
  {
    "category": "Automotive",
    "subcategories": ["Car Repair", "Car Sales", "Tires", "Auto Parts"]
  },
  {
    "category": "Health & Medical",
    "subcategories": [
      "Hospitals",
      "Clinics",
      "Dentists",
      "Pharmacies",
      "Physiotherapy Centres",
      "Veterinary Clinics"
    ]
  },
  {
    "category": "Education",
    "subcategories": [
      "Schools (primary, secondary)",
      "Colleges & Universities",
      "Tutoring Services",
      "Educational Consultants"
    ]
  },
  {
    "category": "Restaurants & Food",
    "subcategories": [
      "Fast Food",
      "Fine Dining",
      "Cafes",
      "Bakeries",
      "Food Delivery Services"
    ]
  },
  {
    "category": "Shopping & Retail",
    "subcategories": [
      "Clothing Stores",
      "Electronics Stores",
      "Jewellery Showrooms",
      "Grocery Stores",
      "Department Stores",
      "Furniture Showrooms",
      "Pest Control"
    ]
  },
  {
    "category": "Professional Business Services",
    "subcategories": [
      "Chartered Accountants",
      "Lawyers",
      "Marketing Agencies",
      "Business Consultants"
    ]
  },
  {
    "category": "Home Services",
    "subcategories": [
      "Interior Designers",
      "Carpenters",
      "Plumbers",
      "Electricians",
      "Cleaners",
      "Landscapers",
      "Home Renovation"
    ]
  },
  {
    "category": "Travel & Transportation",
    "subcategories": [
      "Airlines",
      "Hotels",
      "Car Rentals",
      "Travel Agencies",
      "Public Transport"
    ]
  },
  {
    "category": "Real Estate",
    "subcategories": [
      "Builders and Developers",
      "Civil Contractors",
      "Real Estate Agents",
      "Property Management"
    ]
  },
  {
    "category": "Financial Services",
    "subcategories": [
      "Banks",
      "Loan Agencies",
      "Insurance Agencies",
      "Investment Services",
      "Credit Unions"
    ]
  },
  {
    "category": "Government & Community",
    "subcategories": [
      "Local Government Offices",
      "Community Centers",
      "Libraries",
      "Post Offices"
    ]
  },
  {
    "category": "Entertainment & Media",
    "subcategories": [
      "Movie Theaters",
      "Music Stores",
      "Event Management",
      "Convention Halls",
      "Media Production"
    ]
  },
  {
    "category": "Sports & Recreation",
    "subcategories": [
      "Gyms",
      "Sports Leagues",
      "Parks",
      "Recreation Centers",
      "Sports Equipment Stores"
    ]
  },
  {
    "category": "Pets & Animals",
    "subcategories": [
      "Pet Stores",
      "Veterinarians",
      "Pet Grooming",
      "Animal Shelters",
      "Pet Training"
    ]
  },
  {
    "category": "Computers & Mobile",
    "subcategories": [
      "Internet Service Providers",
      "Computer Sales and Repair",
      "Mobile Sales and Repair",
      "Web Design",
      "Software Solutions"
    ]
  },
  {
    "category": "Safety & Security",
    "subcategories": [
      "Private Security Services",
      "Fire Departments",
      "Police Stations",
      "Safety Equipment"
    ]
  }
]
