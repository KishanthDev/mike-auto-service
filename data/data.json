{
    "businessHours": [
      { "day": "Monday - Friday", "hours": "7:30 AM - 6:00 PM" },
      { "day": "Saturday", "hours": "8:00 AM - 4:00 PM" },
      { "day": "Sunday", "hours": "Closed" }
    ],
    "certifications": ["AUDI", "BMW", "Honda", "Toyota", "Ford"],
    "images": [
      "https://via.placeholder.com/150/92c952",
      "https://via.placeholder.com/150/92c952",
      "https://via.placeholder.com/150/92c952",
      "https://via.placeholder.com/150/92c952"
    ],
    "reviews": [
      {
        "author": "Sarah J.",
        "rating": 5,
        "content": "Mike's team fixed my Honda's transmission issue quickly and for much less than the dealership quoted. Very honest and transparent about costs."
      },
      {
        "author": "David T.",
        "rating": 5,
        "content": "I've been coming here for 5 years and have always had excellent service. They explain everything clearly and never push unnecessary repairs."
      }
    ],
    "services": [
      {
        "title": "Brake Service",
        "description": "Complete brake system inspection, repair and replacement. Lifetime warranty on most brake pads.",
        "price": "Starting at $89.99"
      },
      {
        "title": "Engine Repair",
        "description": "Diagnostic and repair services for all engine types. Factory-trained technicians.",
        "price": "Diagnosis: $89.99"
      },
      {
        "title": "Transmission Service",
        "description": "Transmission fluid change, repair and rebuilding services with 24-month warranty.",
        "price": "Fluid change: $149.99"
      },
      {
        "title": "Electrical Systems",
        "description": "Computer diagnostics, battery service, alternator replacement and more.",
        "price": "Starting at $69.99"
      }
    ]
  }
  