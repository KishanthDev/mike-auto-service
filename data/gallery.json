[
    "https://via.placeholder.com/150/92c952",
    "https://via.placeholder.com/150/92c952",
    "https://via.placeholder.com/150/92c952",
    "https://via.placeholder.com/150/92c952"
  ]
  