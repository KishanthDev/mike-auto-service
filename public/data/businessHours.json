[
    { "day": "Monday - Friday", "hours": "7:30 AM - 6:00 PM" },
    { "day": "Saturday", "hours": "8:00 AM - 4:00 PM" },
    { "day": "Sunday", "hours": "Closed" }
  ]
  