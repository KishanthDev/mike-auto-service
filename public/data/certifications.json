["AUDI", "BMW", "Honda", "Toyota", "Ford"]
