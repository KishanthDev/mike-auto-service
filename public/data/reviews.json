[
    {
      "author": "Sarah J.",
      "rating": 5,
      "content": "Mike's team fixed my Honda's transmission issue quickly and for much less than the dealership quoted. Very honest and transparent about costs."
    },
    {
      "author": "David T.",
      "rating": 5,
      "content": "I've been coming here for 5 years and have always had excellent service. They explain everything clearly and never push unnecessary repairs."
    }
  ]
  