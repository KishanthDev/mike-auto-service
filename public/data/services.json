[
    {
      "title": "Brake Service",
      "description": "Complete brake system inspection, repair and replacement. Lifetime warranty on most brake pads.",
      "price": "Starting at $89.99"
    },
    {
      "title": "Engine Repair",
      "description": "Diagnostic and repair services for all engine types. Factory-trained technicians.",
      "price": "Diagnosis: $89.99"
    },
    {
      "title": "Transmission Service",
      "description": "Transmission fluid change, repair and rebuilding services with 24-month warranty.",
      "price": "Fluid change: $149.99"
    },
    {
      "title": "Electrical Systems",
      "description": "Computer diagnostics, battery service, alternator replacement and more.",
      "price": "Starting at $69.99"
    }
  ]
  