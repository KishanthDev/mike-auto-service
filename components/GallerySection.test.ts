// GallerySection.test.tsx
import React from "react";
import { render, screen } from "@testing-library/react";
import GallerySection from "./GallerySection";
import { GalleryImage } from "@/types/data";

// Proper mock for next/image
jest.mock("next/image", () => {
  return function MockImage(props: any) {
    return React.createElement("img", {
      src: props.src,
      alt: props.alt,
      className: props.className,
      style: {
        width: "100%",
        height: "100%",
        objectFit: "cover",
      },
    });
  };
});

describe("GallerySection Component", () => {
  const mockImages: GalleryImage[] = [
    "/image1.jpg",
    "/image2.jpg",
    "/image3.jpg",
    "/image4.jpg",
    "/image5.jpg",
    "/image6.jpg",
  ];

  beforeEach(() => {
    render(React.createElement(GallerySection, { images: mockImages }));
  });

  it("displays the correct heading", () => {
    const heading = screen.getByRole("heading", { level: 2 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent("Our Shop Gallery");
  });

  it("renders all provided images", () => {
    const images = screen.getAllByRole("img");
    expect(images).toHaveLength(mockImages.length);

    images.forEach((image, index) => {
      expect(image).toHaveAttribute("src", mockImages[index]);
      expect(image).toHaveAttribute("alt", `Gallery ${index + 1}`);
    });
  });

  it("applies correct grid layout classes", () => {
    const grid = screen.getByTestId("gallery-grid");
    expect(grid).toHaveClass("grid");
    expect(grid).toHaveClass("grid-cols-2");
    expect(grid).toHaveClass("sm:grid-cols-4");
  });

  it("applies correct image container styles", () => {
    const containers = screen.getAllByTestId("image-container");
    containers.forEach((container) => {
      expect(container).toHaveClass("aspect-square");
      expect(container).toHaveClass("rounded-lg");
    });
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      React.createElement(GallerySection, { images: mockImages }),
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
