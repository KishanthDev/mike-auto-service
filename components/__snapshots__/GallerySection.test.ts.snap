// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GallerySection Component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h2
      class="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#1a3c6e] dark:text-white"
    >
      Our Shop Gallery
    </h2>
    <div
      class="grid grid-cols-2 sm:grid-cols-4 gap-3 md:gap-4 mt-4 md:mt-6"
      data-testid="gallery-grid"
    >
      <div
        class="aspect-square overflow-hidden rounded-lg bg-gray-100 dark:bg-gray-700 shadow-md"
        data-testid="image-container"
      >
        <img
          alt="Gallery 1"
          class="w-full h-full object-cover transition-transform hover:scale-105"
          src="/image1.jpg"
          style="width: 100%; height: 100%; object-fit: cover;"
        />
      </div>
      <div
        class="aspect-square overflow-hidden rounded-lg bg-gray-100 dark:bg-gray-700 shadow-md"
        data-testid="image-container"
      >
        <img
          alt="Gallery 2"
          class="w-full h-full object-cover transition-transform hover:scale-105"
          src="/image2.jpg"
          style="width: 100%; height: 100%; object-fit: cover;"
        />
      </div>
      <div
        class="aspect-square overflow-hidden rounded-lg bg-gray-100 dark:bg-gray-700 shadow-md"
        data-testid="image-container"
      >
        <img
          alt="Gallery 3"
          class="w-full h-full object-cover transition-transform hover:scale-105"
          src="/image3.jpg"
          style="width: 100%; height: 100%; object-fit: cover;"
        />
      </div>
      <div
        class="aspect-square overflow-hidden rounded-lg bg-gray-100 dark:bg-gray-700 shadow-md"
        data-testid="image-container"
      >
        <img
          alt="Gallery 4"
          class="w-full h-full object-cover transition-transform hover:scale-105"
          src="/image4.jpg"
          style="width: 100%; height: 100%; object-fit: cover;"
        />
      </div>
      <div
        class="aspect-square overflow-hidden rounded-lg bg-gray-100 dark:bg-gray-700 shadow-md"
        data-testid="image-container"
      >
        <img
          alt="Gallery 5"
          class="w-full h-full object-cover transition-transform hover:scale-105"
          src="/image5.jpg"
          style="width: 100%; height: 100%; object-fit: cover;"
        />
      </div>
      <div
        class="aspect-square overflow-hidden rounded-lg bg-gray-100 dark:bg-gray-700 shadow-md"
        data-testid="image-container"
      >
        <img
          alt="Gallery 6"
          class="w-full h-full object-cover transition-transform hover:scale-105"
          src="/image6.jpg"
          style="width: 100%; height: 100%; object-fit: cover;"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
